test/integration/backward_compatible/parallel/serialization/DefaultSerializersLiveTest.js

```diff
@@ -664,3 +658,50 @@ describe('DefaultSerializersLiveTest', function () {
         }
     });
 });
+
+describe('DefaultSerializersLiveTest Arrays', function () {
+    let cluster, client;
+
+    const testFactory = new TestUtil.TestFactory();
+
+    before(async function () {
+        cluster = await testFactory.createClusterForParallelTests();
+        const member = await RC.startMember(cluster.id);
+        client = await testFactory.newHazelcastClientForParallelTests({
+            clusterName: cluster.id,
+            serialization: {
+                defaultNumberType: 'byte'
+            }
+        }, member);
+    });
+
+    beforeEach(async function() {
+        map = await client.getMap(TestUtil.randomString(10));
+    });
+
+    after(async function () {
+        await testFactory.shutdownAll();
+    });
+    it('should get same string values', async function () {
+        await map.put('a', ['a', 'v', 'vg']);
+        const response = await RC.executeOnController(cluster.id, generateGet('a'), Lang.JAVASCRIPT);
+        expect(response.result.toString()).to.equal(['a', 'v', 'vg'].toString());
+    });
+    it('should get values on byte range(-128, +127) when we set numbers out of bounds of byte', async function () {
+        const sampleNumbersArray = [2121, 12121, 547879785, 0, 1, -1, -2121, -12121, -547879785];
+        const expectedReturn =
+            [(2121%128), (12121%128), (547879785%128), (0%128), (1%128), (-1%128), (-2121%128), (-12121%128), (-547879785%128)];
+        const nums = Buffer.from(sampleNumbersArray);
```

Comment:
I think we should not use buffer here. We want to test number serializer. So you can use a for loop to iterate sampleNumbersArray and put one number at a time under different keys. In the loop you can call executeOnController and do the assertion for that key.

Reply:
The same thing applies to the test `should get same byte values`